From 1673c9cb812c967e22d0076ba7691df5bcb48932 Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Thu, 4 Feb 2016 10:37:36 +0000
Subject: [PATCH 03/24] usb: ssic: fix for KW issue 5116007

This patch fixes the KW issue 5116007 in the Intel SSIC GPIO
driver implementation.

Change-Id: I19700e593c2374e9f45130be60e8d90799bdafeb
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18678
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/466443
---
 drivers/platform/x86/intel-gpio-ssic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/intel-gpio-ssic.c b/drivers/platform/x86/intel-gpio-ssic.c
index 0338c05..464d602 100644
--- a/drivers/platform/x86/intel-gpio-ssic.c
+++ b/drivers/platform/x86/intel-gpio-ssic.c
@@ -114,7 +114,7 @@ static int gpio_ssic_request_irq(struct gpio_ssic *ssic)
 	int ret;
 
 	if (ssic->irq_enabled)
-		return ret;
+		return -EBUSY;
 
 	ret = devm_request_threaded_irq(dev, ssic->irq_wakeup, NULL,
 			gpio_ssic_wakeup_irq,
-- 
1.9.1

